src/lib.rs

```diff
@@ -1113,21 +1083,67 @@ pub mod invariant {
         }
 
         #[ink(message)]
-        fn get_liquidity_ticks_amount(&self, pool_key: PoolKey) -> u32 {
+        fn get_liquidity_ticks_amount(
+            &self,
+            pool_key: PoolKey,
+            lower_tick: i32,
+            upper_tick: i32,
+        ) -> Result<u32, InvariantError> {
             let tick_spacing = pool_key.fee_tier.tick_spacing;
+            if tick_spacing == 0 {
+                return Err(InvariantError::InvalidTickSpacing);
+            };
+
+            if lower_tick % (tick_spacing as i32) != 0 || upper_tick % (tick_spacing as i32) != 0 {
+                return Err(InvariantError::InvalidTickIndex);
+            }
 
             let max_tick = get_max_tick(tick_spacing);
-            let (chunk_limit, _) = tick_to_position(max_tick, tick_spacing);
+            let min_tick = get_min_tick(tick_spacing);
+
+            if lower_tick < min_tick || upper_tick > max_tick {
+                return Err(InvariantError::InvalidTickIndex);
+            };
+
+            let (min_chunk_index, min_bit) = tick_to_position(lower_tick, tick_spacing);
+            let (max_chunk_index, max_bit) = tick_to_position(upper_tick, tick_spacing);
+
+            let active_bits_in_range = |chunk, min_bit, max_bit| {
+                let mut amount = 0;
+                for bit in min_bit..=max_bit {
+                    amount += ((chunk >> bit) & 1) as u32;
+                }
+                amount
+            };
+
+            let min_chunk = self
+                .tickmap
+                .bitmap
+                .get((min_chunk_index, pool_key))
+                .unwrap_or(0);
+
+            if max_chunk_index == min_chunk_index {
+                return Ok(active_bits_in_range(min_chunk, min_bit, max_bit));
+            }
+
+            let max_chunk = self
+                .tickmap
+                .bitmap
+                .get((max_chunk_index, pool_key))
+                .unwrap_or(0);
+
+            let mut amount: u32 = 0;
 
-            let mut amount = 0;
+            amount += active_bits_in_range(min_chunk, min_bit, 63);
```

Comment:
Here you can even use `CHUNK_SIZE - 1` const instead magic number (just for code clarity)